CodeU_Assignement_5/src/codeu_assignement_5/kemUnknownLanguage.java

```diff
@@ -0,0 +1,177 @@
+        {
+            /* Get the current order by prefix size, starting from 0 */
+            dict = getBasicAlphaBeit(alienWords, 0); /* Basic Alphabeit */ 
+            /* A,R,C from the example! */
+            
+            /* Add from prefix while there are letters after first letter of prefix */
+            while (!addToDictByPrefix(alienWords, dict)){} 
+            
+            /* Add all the chars we cannot decide in the end*/
+            for (String w : alienWords) 
+            {
+                for (Character c: w.toCharArray())
+                {
+                    if (!dict.contains(c))
+                    {
+                        dict.add(c);
+                    }
+                }
+            }
+        }
+        
+        return dict;
+    }
+    
+    /*
+     * Inputs: All words with the same prefix (all the first prefixSize chars are the same!)
+     * Output: Per common prefix, gets the order of the chars
+     *
+     * Note: Assume the input is valid here!
+    */
+    private List<Character> getBasicAlphaBeit(String[] words, int prefixSize)
+    {
+        List<Character> ret = new ArrayList(); 
+        if (words == null)
+        {
+            return ret;
+        }
+        
+        for (String w : words) 
+        {
+            if (!ret.contains(w.charAt(prefixSize)))
+            {
+                ret.add(w.charAt(prefixSize));
+            }
+        }   
+        
+        return ret;
+    }
+    
+    /*
+     * Gives the prefix of a char that isn't in the dictionary
+    */
+    private int getSizeOfPrefix(String w, List<Character> dict)
+    {
+        if ((w == null) || (w.length() == 0) || (dict == null) || (dict.isEmpty()))
+        {
+            return 0;
+        }
+        
+        for (int i=0; i < w.length(); i++)
+        {
+            if (!dict.contains(w.charAt(i)))
+            {
+                return i;
+            }
+        }
+        
+        return w.length();
+    }
+    
+    /*
+     * Input: the original set of words, current word we are working on, its prefix size
+     * Output: the set of the words with this prefix
+    */
+    private String[] getWordsWithSamePrefix(String[] alienWords, int curr, int prefixSize)
+    {
+        ArrayList<String> tempDict = new ArrayList();
+        tempDict.add(alienWords[curr]);
+        
+        String prefix = alienWords[curr].substring(0, prefixSize);
+        
+        /* Forward search */
+        for (int i=curr+1; i<alienWords.length; i++)
+        {
+            if (!alienWords[i].startsWith(prefix))
+            {
+                break;
+            }
+            
+            tempDict.add(alienWords[i]);
+        }
+        
+        /* Backward search */
+        for (int i=curr-1; i >= 0; i--)
+        {
+            if (!alienWords[i].startsWith(prefix))
+            {
+                break;
+            }
+            
+            tempDict.add(0,alienWords[i]); 
+        }

```

Comment:
oh no, just a personal vendetta against break that's it :) so happy if you just not agree :)